cpu/sam0_common/periph/adc.c

```diff
@@ -87,67 +187,43 @@ static int _adc_configure(adc_res_t res)
         return -1;
     }
     _adc_poweroff();
-    if (ADC->CTRLA.reg & ADC_CTRLA_SWRST ||
-        ADC->CTRLA.reg & ADC_CTRLA_ENABLE ) {
+    if (ADC_DEV->CTRLA.reg & ADC_CTRLA_SWRST ||
+        ADC_DEV->CTRLA.reg & ADC_CTRLA_ENABLE ) {
         DEBUG("adc: not ready\n");
         return -1;
     }
-#ifdef CPU_SAMD21
-    /* Power On */
-    PM->APBCMASK.reg |= PM_APBCMASK_ADC;
-    /* GCLK Setup */
-    GCLK->CLKCTRL.reg = (uint32_t)(GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 |
-                        (GCLK_CLKCTRL_ID(ADC_GCLK_ID)));
-    /* Configure CTRLB Register HERE IS THE RESOLUTION SET! */
-    ADC->CTRLB.reg = ADC_PRESCALER | res;
-    /* Load the fixed device calibration constants */
-    ADC->CALIB.reg =
-        ADC_CALIB_BIAS_CAL((*(uint32_t*)ADC_FUSES_BIASCAL_ADDR >>
-                            ADC_FUSES_BIASCAL_Pos)) |
-        ADC_CALIB_LINEARITY_CAL((*(uint64_t*)ADC_FUSES_LINEARITY_0_ADDR >>
-                                ADC_FUSES_LINEARITY_0_Pos));
+    _setup_clock();
+    _setup_calibration();
+
+    /* Set ADC resolution */
+#ifdef ADC_CTRLC_RESSEL
+    /* Reset resolution bits in CTRLC */
+    ADC_DEV->CTRLC.reg &= ~ADC_CTRLC_RESSEL_Msk;
+    ADC_DEV->CTRLC.reg |= res;
+#else
+    ADC_DEV->CTRLB.reg |= res;
```

Comment:
You want to do the same here

Reply:
No, we set CTRLB explicitly to `ADC_PRESCALER` in `_setup_clock`. It's not necessary here.

Reply:
True. Thought it is not necessary because we set CTRLB to `ADC_PRESCALER` explicitly in `_setup_clock`. But that's just for those CPUs with CTRLC for the resolution. Missed that one.